apps/emqx/src/emqx_persistent_session_ds/emqx_persistent_session_ds_shared_subs.erl

```diff
@@ -324,48 +324,53 @@ schedule_unsubscribe(
 %%--------------------------------------------------------------------
 %% pre_renew_streams
 
--spec pre_renew_streams(emqx_persistent_session_ds_state:t(), t()) ->
-    {emqx_persistent_session_ds_state:t(), t()}.
-pre_renew_streams(S, SharedSubS) ->
-    on_streams_replay(S, SharedSubS).
+-spec pre_renew_streams(
+    emqx_persistent_session_ds_state:t(), t(), emqx_persistent_session_ds_stream_scheduler:t()
+) ->
+    {emqx_persistent_session_ds_state:t(), t(), emqx_persistent_session_ds_stream_scheduler:t()}.
+pre_renew_streams(S, SharedSubS, SS) ->
+    on_streams_replay(S, SharedSubS, SS).
 
 %%--------------------------------------------------------------------
 %% renew_streams
 
--spec renew_streams(emqx_persistent_session_ds_state:t(), t()) ->
-    {emqx_persistent_session_ds_state:t(), t()}.
-renew_streams(S0, #{agent := Agent0, scheduled_actions := ScheduledActions} = SharedSubS0) ->
+-spec renew_streams(
+    emqx_persistent_session_ds_state:t(), t(), emqx_persistent_session_ds_stream_scheduler:t()
+) ->
+    {emqx_persistent_session_ds_state:t(), t(), emqx_persistent_session_ds_stream_scheduler:t()}.
+renew_streams(S0, #{agent := Agent0, scheduled_actions := ScheduledActions} = SharedSubS0, SchedS) ->
     {StreamLeaseEvents, Agent1} = emqx_persistent_session_ds_shared_subs_agent:renew_streams(
         Agent0
     ),
     StreamLeaseEvents =/= [] andalso
         ?tp(warning, shared_subs_new_stream_lease_events, #{
             stream_lease_events => format_lease_events(StreamLeaseEvents)
         }),
-    S1 = lists:foldl(
+    {S1, NewSchedS} = lists:foldl(
         fun
-            (#{type := lease} = Event, S) -> accept_stream(Event, S, ScheduledActions);
-            (#{type := revoke} = Event, S) -> revoke_stream(Event, S)
+            (#{type := lease} = Event, {S, SS}) -> accept_stream(Event, S, ScheduledActions, SS);
+            %%@TODO
```

Comment:
What else should be done for revoking? Will poll results be just ignored for unsubscribed streams?

Reply:
I feel it will be some race condition for DS  since revoking is async.  I didn't find a test case testing that so I added @TODO for double check.

Reply:
we down prio the DS shard sub, so this is no longer a merge blocker.